### PR TITLE
SDKv2: IDP changes/tweaks

### PR DESCRIPTION
**Pull request by sagidayan**

### CODE REVIEW

File: internal/rhcs/idps/identity_provider_state.go [hunk 1]
```diff
@@ -50,7 +51,8 @@ func IdentityProviderFields() map[string]*schema.Schema {
 			Elem: &schema.Resource{
 				Schema: HtpasswdSchema(),
 			},
-			Optional: true,
+			Optional:     true,
+			ExactlyOneOf: []string{"htpasswd", "gitlab", "github", "google", "ldap", "openid"},
```
Comment (osherdp) [inline internal/rhcs/idps/identity_provider_state.go:55]: maybe let's have a constant for ``[]string{"htpasswd", "gitlab", "github", "google", "ldap", "openid"}``?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: internal/rhcs/idps/identity_provider_state.go (hunk 1 above)
Comment (sagidayan) [reply internal/rhcs/idps/identity_provider_state.go]: Updated

**Review by osherdp — APPROVED** [review]

**Comment by nirarg** [comment]
Closing this PR as it was decided to move to latest framework

**Pull request closed without merging**